add trigger_helm_update var & add data block

sachincool opened the pull request.

sachincool committed:
refactor(terraform): replace token variable with aws_eks_cluster_auth data source for authentication and update cluster_name variable for clarity

sachincool committed:
feat(terraform): add trigger_helm_update variable to control Helm chart updates

Introduce a new variable `trigger_helm_update` to allow users to
trigger Helm chart updates. This enhances flexibility by enabling
conditional updates based on the variable's value.

github-actions[bot] committed:
terraform-docs: automated action

dunefro commented on main.tf:
```diff
@@ -1,9 +1,13 @@
+    chart_version  = var.chart_version
+    release_name   = var.release_name
+    namespace      = var.namespace
+    update_trigger = var.trigger_helm_update != null ? var.trigger_helm_update : "initial"
```
this doesn't ensure it will trigger always - the value of `var.trigger_helm_update` will be fixed. it should be something like
```
update_trigger = var.trigger_helm_update != null ? timestamp() : "initial"
```

sachincool replied:
makes it much better, I like this

dunefro commented on main.tf:
```diff
@@ -32,7 +36,7 @@ resource "null_resource" "helm_install" {
       - name: aws
         user:
-          token: ${var.token}
+          token: ${data.aws_eks_cluster_auth.cluster.token}
```
Why can't we get the token from the output of eks module. Its not super important but the terraform module should keep only the things specific to itself.

Also is this done because the token expires ?

dunefro replied:
we should also derive `${var.cluster_endpoint}` and `${var.cluster_ca_certificate}` from the same data source

sachincool replied:
> Why can't we get the token from the output of eks module. Its not super important but the terraform module should keep only the things specific to itself.

I checked before, token is not passed in outputs of eks module, neither from us, and nor from aws module

sachincool replied:
we can use this
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#attribute-reference
 for fetching `cluster_endpoint` and `ca_certificate`, 

 doing that 
 once in main.tf -> passing it in helm
 
Or 

do data block call in helm everytime we use helm 

P.s We're also making data block part of modules.yaml,  similar to modules section, we will have a data block section in yaml, 
since, we also require aws data block to fetch account id from current profile. ( making the end user do less work)

dunefro commented on variables.tf:
```diff
@@ -44,13 +44,18 @@ variable "cluster_endpoint" {
+}
+variable "trigger_helm_update" {
+  description = "Set this to any value to trigger a Helm chart update"
+  type        = string
```
Let's make it bool and check my above comment

dunefro commented on main.tf:
```diff
@@ -1,9 +1,13 @@
+data "aws_eks_cluster_auth" "cluster" {
+  name = var.cluster_name
+}
```
Just add a new line after this - to keep things clean

sachincool committed:
changed update_trigger to bool

github-actions[bot] committed:
terraform-docs: automated action

sachincool committed:
refactor(terraform): use aws_eks_cluster data source for dynamic cluster info

Replace hardcoded cluster endpoint and CA certificate variables with
dynamic retrieval using aws_eks_cluster data source. This change
removes the need for cluster_ca_certificate and cluster_endpoint
variables, simplifying configuration and reducing manual input.

github-actions[bot] committed:
terraform-docs: automated action

dunefro reviewed (approved):
LGTM

sachincool committed:
refactor(main.tf): streamline KUBECONFIG handling by using export to simplify Helm command execution and cleanup

sachincool committed:
docs(variables.tf): clarify description for trigger_helm_update variable to specify it should be set to true to trigger an update

github-actions[bot] committed:
terraform-docs: automated action
